sys/vm/vm_map.c

```diff
@@ -3337,9 +3337,8 @@ vm_map_protect(vm_map_t map, vm_offset_t start, vm_offset_t end,
 		 * potential compatibility breakage.  The alternative
 		 * would maximize compatibility.
 		 */
-		if (flags & VM_MAP_PROTECT_KEEP_CAP &&
-		    (((flags & VM_MAP_PROTECT_SET_MAXPROT) != 0 && (entry->max_protection & VM_PROT_CAP) != 0) ||
-		    ((flags & VM_MAP_PROTECT_SET_MAXPROT) == 0 && (entry->protection & VM_PROT_CAP) != 0))) {
+		if ((flags & VM_MAP_PROTECT_KEEP_CAP) != 0 &&
```

Comment:
I believe this is ok to only check max_protection.  In particular, entry->protection can only be a subset, so there's no case where entry->protection would contain some VM_PROT_CAP bits and entry->max_protection would not.